lib/puppet/type/rpmkey.rb

```diff
@@ -23,4 +23,8 @@
     desc "The source of the public key if the key is not already imported."
   end
 
+  autorequire(:file) do
+    self[:source] if self[:source] =~ /^\//
```

Comment:
we could also use `Puppet::Util.absolute_path?` which is supposed to be platform agnostic, but since this type does not make sense on windows, this will do as well